### ISSUE TITLE
Problems with "Umlaute"

### ISSUE THREAD

**Issue by mikolama** [marytts/marytts#292]
I'm new to MaryTTS and just started experimenting with it. I'm running a local Mary 5.1.2 server and tested several voices with the "official" marytts-client and the MaryInterface Java-API. This works fine but sometimes with some words, containing some "Umlaute" or their substitutes (e.g. 'ä' -> 'ae'), the application crashes with a "marytts.exceptions.SynthesisException".

The example was tested both ways - in my own Java-Application and in the included marytts-client.

Example:
The sentence "*Schöner Tag.*" is pronounced correctly and outputted without error.
The input "*Gepäck*" and "*Gepaeck*" terminates the program with following exceptions: 

<html><body><pre>Exception in thread "main" marytts.exceptions.SynthesisException: java.io.IOException: Error message from server:
Internal server error Processing failed. 
java.lang.Exception: Module JPhonemiser_de: Problem processing the data.
	at marytts.server.Request.processOneChunk(Request.java:549)
	at marytts.server.Request.processOrLookupOneChunk(Request.java:385)
	at marytts.server.Request.process(Request.java:327)
	at marytts.server.http.SynthesisRequestHandler.process(SynthesisRequestHandler.java:261)
	at marytts.server.http.SynthesisRequestHandler.handleClientRequest(SynthesisRequestHandler.java:91)
	at marytts.server.http.BaseHttpRequestHandler.handle(BaseHttpRequestHandler.java:138)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler$RequestHandlerAdaptor.handle(BufferingHttpServiceHandler.java:189)
	at org.apache.http.nio.protocol.SimpleNHttpRequestHandler.handle(SimpleNHttpRequestHandler.java:51)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.processRequest(AsyncNHttpServiceHandler.java:453)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.inputReady(AsyncNHttpServiceHandler.java:316)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler.inputReady(BufferingHttpServiceHandler.java:131)
	at org.apache.http.impl.nio.DefaultNHttpServerConnection.consumeInput(DefaultNHttpServerConnection.java:173)
	at org.apache.http.impl.nio.DefaultServerIOEventDispatch.inputReady(DefaultServerIOEventDispatch.java:147)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:161)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:335)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:275)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:542)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: Allophone `-' could not be found in AllophoneSet `sampa' (Locale: de)
	at marytts.modules.phonemiser.AllophoneSet.getAllophone(AllophoneSet.java:224)
	at marytts.modules.phonemiser.Syllabifier.correctStressSymbol(Syllabifier.java:297)
	at marytts.modules.phonemiser.Syllabifier.syllabify(Syllabifier.java:252)
	at marytts.modules.phonemiser.Syllabifier.syllabify(Syllabifier.java:52)
	at marytts.modules.phonemiser.TrainedLTS.syllabify(TrainedLTS.java:166)
	at marytts.language.de.JPhonemiser.phonemise(JPhonemiser.java:349)
	at marytts.language.de.JPhonemiser.process(JPhonemiser.java:232)
	at marytts.server.Request.processOneChunk(Request.java:547)
	... 19 more

	at marytts.client.RemoteMaryInterface.generateAudio(RemoteMaryInterface.java:250)
	at _______TESTS.MaryClientUser.main(MaryClientUser.java:118)
Caused by: java.io.IOException: Error message from server:

Internal server error. Processing failed.
java.lang.Exception: Module JPhonemiser_de: Problem processing the data.
	at marytts.server.Request.processOneChunk(Request.java:549)
	at marytts.server.Request.processOrLookupOneChunk(Request.java:385)
	at marytts.server.Request.process(Request.java:327)
	at marytts.server.http.SynthesisRequestHandler.process(SynthesisRequestHandler.java:261)
	at marytts.server.http.SynthesisRequestHandler.handleClientRequest(SynthesisRequestHandler.java:91)
	at marytts.server.http.BaseHttpRequestHandler.handle(BaseHttpRequestHandler.java:138)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler$RequestHandlerAdaptor.handle(BufferingHttpServiceHandler.java:189)
	at org.apache.http.nio.protocol.SimpleNHttpRequestHandler.handle(SimpleNHttpRequestHandler.java:51)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.processRequest(AsyncNHttpServiceHandler.java:453)
	at org.apache.http.nio.protocol.AsyncNHttpServiceHandler.inputReady(AsyncNHttpServiceHandler.java:316)
	at org.apache.http.nio.protocol.BufferingHttpServiceHandler.inputReady(BufferingHttpServiceHandler.java:131)
	at org.apache.http.impl.nio.DefaultNHttpServerConnection.consumeInput(DefaultNHttpServerConnection.java:173)
	at org.apache.http.impl.nio.DefaultServerIOEventDispatch.inputReady(DefaultServerIOEventDispatch.java:147)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.readable(BaseIOReactor.java:161)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvent(AbstractIOReactor.java:335)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processEvents(AbstractIOReactor.java:315)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:275)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:542)
	at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: Allophone `-' could not be found in AllophoneSet `sampa' (Locale: de)
	at marytts.modules.phonemiser.AllophoneSet.getAllophone(AllophoneSet.java:224)
	at marytts.modules.phonemiser.Syllabifier.correctStressSymbol(Syllabifier.java:297)
	at marytts.modules.phonemiser.Syllabifier.syllabify(Syllabifier.java:252)
	at marytts.modules.phonemiser.Syllabifier.syllabify(Syllabifier.java:52)
	at marytts.modules.phonemiser.TrainedLTS.syllabify(TrainedLTS.java:166)
	at marytts.language.de.JPhonemiser.phonemise(JPhonemiser.java:349)
	at marytts.language.de.JPhonemiser.process(JPhonemiser.java:232)
	at marytts.server.Request.processOneChunk(Request.java:547)
	... 19 more

	at marytts.client.http.MaryHttpClient.requestInputStream(MaryHttpClient.java:397)
	at marytts.client.http.MaryHttpClient._process(MaryHttpClient.java:439)
	at marytts.client.MaryClient.process(MaryClient.java:697)
	at marytts.client.RemoteMaryInterface.processStringToBytes(RemoteMaryInterface.java:269)
	at marytts.client.RemoteMaryInterface.generateAudio(RemoteMaryInterface.java:247)
	... 1 more
Caused by: java.io.IOException: Server returned HTTP response code: 500 for URL: http://localhost:59125/process
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1838)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1439)
	at marytts.client.http.MaryHttpClient.requestInputStream(MaryHttpClient.java:388)
	... 5 more
</pre></body></html>

**Comment by Munzey** [comment]
looks like more regressions from open issue #285 . Currently working on fix!

**Comment by Munzey** [comment]
like with english, we need to rebuild the .fst and .lts files, will do this in a 'rebuild-german' branch

**Comment by psibre** [comment]
Fixed along with #341.

Now we get:

### TEXT
```
Gepäck
Gepaeck
```

### PHONEMES
```xml
<?xml version="1.0" encoding="UTF-8"?><maryxml xmlns="http://mary.dfki.de/2002/MaryXML" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="0.5" xml:lang="de">
<p>
<s>
<t g2p_method="rules" ph="' g @ - p E k" pos="NN">
Gepäck
</t>
<t g2p_method="rules" ph="g @ - p a - ' E k" pos="NN">
Gepaeck
</t>
</s>
</p>
</maryxml>
```
